pwndbg/gdblib/memory.py

```diff
@@ -360,3 +360,55 @@ def update_min_addr() -> None:
     global MMAP_MIN_ADDR
     if pwndbg.gdblib.qemu.is_qemu_kernel():
         MMAP_MIN_ADDR = 0
+
+
+def fetch_struct_as_dictionary(
+    struct_name: str,
+    struct_address: int,
+    include_only_fields: set[str] = set(),
+    exclude_fields: set[str] = set(),
+):
+    struct_type = gdb.lookup_type("struct " + struct_name)
+    fetched_struct = poi(struct_type, struct_address)
+
+    return pack_struct_into_dictionary(fetched_struct, include_only_fields, exclude_fields)
+
+
+def pack_struct_into_dictionary(
+    fetched_struct: gdb.Value,
+    include_only_fields: set[str] = set(),
+    exclude_fields: set[str] = set(),
+):
+    struct_as_dictionary = {}
+
+    if len(include_only_fields) != 0:
+        for field_name in include_only_fields:
+            key = field_name
+            value = convert_gdb_value_to_python_value(fetched_struct[field_name])
+            struct_as_dictionary[key] = value
+    else:
+        for field in fetched_struct.type.fields():
+            if field.name is None:
+                # Flatten anonymous structs/unions
+                struct_as_dictionary.update(
+                    convert_gdb_value_to_python_value(fetched_struct[field])
+                )
+            elif field.name not in exclude_fields:
+                key = field.name
+                value = convert_gdb_value_to_python_value(fetched_struct[field])
+                struct_as_dictionary[key] = value
+
+    return struct_as_dictionary
+
+
+def convert_gdb_value_to_python_value(gdb_value: gdb.Value):
+    type_code_conversions = {
```

Comment:
This dict is build on each function call. Let's move it outside of this function so it is created only once in the module and re-used.

Also do we really need this dict at all? There aren't many options, I know we want to be lazy, but maybe it'd be better to do `if elif ..`?

Reply:
Yeah perhaps I was trying to be terse when I used the dict here, I'm happy to simplify it which could also alleviate the type checker issue. I'll wait on @gsingh93 to see what's going on with that Ubuntu 20.04 test first because it looks related.